be/src/olap/generic_iterators.cpp

```diff
@@ -0,0 +1,346 @@
+    int row_idx = 0;
+    while (row_idx < block->capacity() && _rows_returned < _num_rows) {
+        RowBlockRow row = block->row(row_idx);
+
+        for (int i = 0; i < _schema.column_schemas().size(); ++i) {
+            row.set_is_null(i, false);
+            auto& col_schema = _schema.column_schemas()[i];
+            switch (col_schema.type()) {
+            case OLAP_FIELD_TYPE_SMALLINT:
+                *(int16_t*)row.cell_ptr(i) = _rows_returned + i;
+                break;
+            case OLAP_FIELD_TYPE_INT:
+                *(int32_t*)row.cell_ptr(i) = _rows_returned + i;
+                break;
+            case OLAP_FIELD_TYPE_BIGINT:
+                *(int64_t*)row.cell_ptr(i) = _rows_returned + i;
+                break;
+            case OLAP_FIELD_TYPE_FLOAT:
+                *(float*)row.cell_ptr(i) = _rows_returned + i;
+                break;
+            case OLAP_FIELD_TYPE_DOUBLE:
+                *(double*)row.cell_ptr(i) = _rows_returned + i;
+                break;
+            default:
+                break;
+            }
+        }
+        row_idx++;
+        _rows_returned++;
+    }
+    block->resize(row_idx);
+    if (row_idx > 0) {
+        return Status::OK();
+    }
+    return Status::EndOfFile("End of AutoIncrementIterator");
+}
+
+// Used to store merge state for a MergeIterator input.
+// This class will iterate all data from internal iterator
+// through client call advance().
+// Usage:
+//      MergeContext ctx(iter);
+//      RETURN_IF_ERROR(ctx.init());
+//      while (ctx.valid()) {
+//          visit(ctx.current_row());
+//          RETURN_IF_ERROR(ctx.advance());
+//      }
+class MergeContext {
+public:
+    // This class don't take iter's ownership, client should delete it
+    MergeContext(RowwiseIterator* iter)
+        : _iter(iter), _block(iter->schema(), 1024, &_arena) {
+    }
+
+    // Intialize this context and will prepare data for current_row()
+    Status init(const StorageReadOptions& opts);
+
+    // Return current row which internal row index points to
+    // And this function won't make internal index advance.
+    // Before call this function, Client must assure that
+    // has_remaining() return true
+    RowBlockRow current_row() const {
+        return RowBlockRow(&_block, _index_in_block);
+    }
+
+    // Advance internal row index to next valid row
+    // Return error if error happens
+    // Don't call this when valid() is false, action is undefined
+    Status advance();
+
+    // Return if has remaining data in this context.
+    // Only when this function return true, current_row()
+    // will return a valid row
+    bool valid() const { return _valid; }
+
+private:
+    // Load next block into _block
+    Status _load_next_block();
+
+private:
+    RowwiseIterator* _iter;
+    Arena _arena;
+    // used to store data load from iterator
+    RowBlockV2 _block;
+
+    bool _valid = false;
+    size_t _index_in_block = 0;
```

Comment:
Block has the position, _index_in_block may be redundant.

Reply:
In RowBlockV2, there is no position.